### ISSUE TITLE
Tick Label Width is 0 for new plot

### ISSUE THREAD

**Issue by DavidBaddeley** [flot/flot#1014]
I have just migrated from 0.7 to 0.8 and run into a problem

Whilst generating the tick labels the source code uses element.outerWidth(true) and element.outerHeight(true) to get the dimensions of each tick label, directly after they are added to the DOM, but I am finding that in some cases (on some plots but not others) the .outerWidth(true) returns 0, this results in the tick labels sitting to the right of the y1 axis, on the grid.

Then if the same plot is re-drawn (without refreshing the page) the tick labels are fine

I have tried .width() and its the same again, after some research I found similar behaviour described here

http://stackoverflow.com/questions/6132141/jquery-why-does-width-sometimes-return-0-after-inserting-elements-with-html

Where adding a setTimeout() works, this does seem to work but I wanted to see if there is another way around this as I didnt want to modify the source. Another fix could be to calculate the width based on the length of the text string, but again this isn't ideal

I am using jQuery 1.8.3

The code in questions....

```
Canvas.prototype.getTextInfo = function(layer, text, font, angle) {

    var textStyle, layerCache, styleCache, info;

    // Cast the value to a string, in case we were given a number or such

    text = "" + text;

    // If the font is a font-spec object, generate a CSS font definition

    if (typeof font === "object") {
        textStyle = font.style + " " + font.variant + " " + font.weight + " " + font.size + "px/" + font.lineHeight + "px " + font.family;
    } else {
        textStyle = font;
    }

    // Retrieve (or create) the cache for the text's layer and styles

    layerCache = this._textCache[layer];

    if (layerCache == null) {
        layerCache = this._textCache[layer] = {};
    }

    styleCache = layerCache[textStyle];

    if (styleCache == null) {
        styleCache = layerCache[textStyle] = {};
    }

    info = styleCache[text];

    // If we can't find a matching element in our cache, create a new one

    if (info == null) {

        var element = $("<div></div>").html(text)
            .css({
                position: "absolute",
                top: -9999
            })
            .appendTo(this.getTextLayer(layer))

        if (typeof font === "object") {
            element.css({
                font: textStyle,
                color: font.color
            });
        } else if (typeof font === "string") {
            element.addClass(font);
        }
        var width = element.outerWidth(true);
        if(width<1){
            console.log("bah")
        }
        info = styleCache[text] = {
            active: false,
            rendered: false,
            element: element,
            width: element.outerWidth(true),
            height: element.outerHeight(true)
        };
        element.detach();
    }
    return info;
};
```

**Comment by Nuranto** [comment]
Is it still unresolvable ?
Can't this be done via MutationObserver (cf. https://stackoverflow.com/questions/20487472/jquery-when-element-becomes-visible ?) ?
I did'nt have this kind of issue when using Chart.js for smaller needs. How do they solve this ?
